### PR TITLE
TE: adding anomaly merge component

### PR DESCRIPTION
**Pull request by puneetjaiswal**
adding anomaly merge component

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/merge/AnomalyMergeConfig.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package com.linkedin.thirdeye.anomaly.merge;
+
+public class AnomalyMergeConfig {
+  AnomalyMergeStrategy mergeStrategy = AnomalyMergeStrategy.FUNCTION;
```
Comment (npawar) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/merge/AnomalyMergeConfig.java]: sequentialAllowedGap is the gap which if exceeded, the anomalies don't need to be merged?
And is mergeDuration the max duration for which anomalies can be merged?
Can we add javadoc/comments to explain these?

Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/merge/AnomalyMergeConfig.java]: yes. I will add the javadoc in next commit.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/merge/AnomalySummaryGenerator.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+        // compare current with merged and decide whether to merge the current result or create a new one
+        if (applySequentialGapBasedSplit
+            && (currentResult.getStartTimeUtc() - mergedAnomaly.getEndTime()) > mergeConfig
+            .getSequentialAllowedGap()) {
+          // Split here if not processed already
+          mergedAnomalies.add(mergedAnomaly);
+          mergedAnomaly = new MergedAnomalyResult();
+          populateMergedResult(mergedAnomaly, currentResult);
+          mergedAnomaly.setMergeStrategy(mergeConfig.getMergeStrategy());
```
Comment (npawar) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/merge/AnomalySummaryGenerator.java]: isn't the merge strategy being taken into account anywhere? it looks like a pure time based merge, or am i missing something

Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/merge/AnomalySummaryGenerator.java]: Merge strategy will be used outside this Util, a place where List of AnomalyResult fetched.

**Comment by npawar** [comment]
LGTM

**Pull request merged**